iceoryx_posh/doc/chunk_header.md

```diff
@@ -0,0 +1,218 @@
+```
+sizeLeftOfThePayloandAlignmentBoundary = sizeof(chunkHeader) - alignof(chunkHeader);
+chunkSize = sizeLeftOfThePayloandAlignmentBoundary + payloadAlignment + payloadSize;
+```
+
+3. Custom header is used
+
+Similar to case 2, but in this case it is the `back-offset` which might cross the payload alignment boundary.
+```
+headerSize = sizeof(chunkHeader) + sizeof(customHeader)
+sizeLeftOfThePayloandAlignmentBoundary = align(headerSize, alignof(payloadOffset));
+maxAlignment = max(alignof(payloadOffset), payloadAlignment);
+chunkSize = sizeLeftOfThePayloandAlignmentBoundary + maxAlignment + payloadSize;
+```
+
+### Accessing Chunk Header Extension
+
+The `ChunkHeader` has a template method to get custom header. There is a risk to use this wrong by accident, but there is currently no better solution for this.
+
+Since the custom header is always adjacent to the `ChunkHeader`, the formula to get the custom header is:
+```
+customHeader = addressOf(chunkHeader) + sizeof(chunkHeader);
+```
+
+### ChunkHeader Methods
+
+- `void* payload()` returns a pointer to the payload
+- `template <typename T> T* customHeader()` returns a pointer to the custom header
+- `static ChunkHeader* fromPayload(const void* const payload)` returns a pointer to the `ChunkHeader` associated to the payload
+
+### Integration Into Publisher/Subscriber API
+
+The `Publisher` has additional template parameters for the custom header and payload alignment. Alternatively this could also be done with the `allocate` method, but that increases the risk of using it wrong by accident.
+
+Additionally, to fully integrate the custom header into the regular control flow, a pointer to a `ChunkHeaderHook` can be passed to the `Publisher` c'tor.
+
+The `ChunkHeaderHook` is not in the shared memory and has the following virtual methods:
+- `virtual void allocateHook(ChunkHeader& chunkHeader)` can be used to initialize the custom header
+- `virtual void deliveryHook(ChunkHeader& chunkHeader)` can be used to set members of the custom header, e.g. a timestamp
+
+Furthermore, the `Publisher` and `Subscriber` have access to the `ChunkHeader` and can use the `customHeader()` method to gain access to the custom header.
+
+### Pitfalls & Testing
+
+- when the payload is adjacent to the `ChunkHeader`, it must be ensured that `m_payloadOffset` overlaps with the `back-offset`, which is `sizeof(m_payloadOffset)` in front of the payload
+- to simplify calculation, it is assumed that the alignment of the custom header doesn't exceed the alignment of the `ChunkHeader`. This has to be enforced with an `assert`
+
+# Open Issues
+
+- the design was done with the intention to have a custom header of arbitrary size, if the size is limited to e.g. 32 bytes, some things could be simplified
+- untyped API and C bindings
+    - PoC is needed
+- user defined sequence number
+    - this can probably be done by a `ChunkHeaderHook`
```

Comment:
Then the user defined sequence number would be part of the custom chunk header. Maybe this is the best for now. If iceoryx is used in a framework that wants to have additional features based on the sequence number having an additional custom chunk header there is quite likely (e.g. there are timestamps etc.)

Reply:
either that or the `ChunkHeaderHook`could also be used without a custom header, to overwrite the sequence number in `ChunkHeader`. I'm not yet sure if this is a good idea though. maybe we don't want to give the user write access to the `ChunkHeader`

Reply:
I would only allow access to the custom chunk header in the hook. The chunk header is then a header only iceoryx writes. Having these separation of concerns makes sense for me when we have a custom header